lib/vega_lite/data.ex

```diff
@@ -403,42 +403,58 @@ defmodule VegaLite.Data do
         legend_key: legend_key
       )
   """
-  def polar_plot(vl \\ Vl.new(), data, fields, opts \\ []) do
+  def polar_grid(vl \\ Vl.new(), opts \\ []) do
     opts =
       Keyword.validate!(
         opts,
         [
           :radius_marks,
-          :mark,
-          mark_opts: [],
-          return_grid: true,
-          direction: :counter_clockwise,
           angle_marks: [0, 90, 180, 270, 360],
+          direction: :counter_clockwise,
           angle_offset: 0,
-          grid_opacity: 1,
-          grid_color: "white",
-          grid_stroke_color: "black",
-          grid_stroke_opacity: 1,
-          scheme: "turbo"
+          opacity: 1,
+          color: "white",
+          stroke_color: "black",
+          stroke_opacity: 1
         ]
       )
 
-    data_layer = polar_plot_data_layers(data, fields, opts[:mark], opts[:mark_opts], opts)
+    unless opts[:radius_marks] do
+      raise ArgumentError, "missing :radius_marks option"
+    end
+
+    grid_layers = polar_angle_layers(opts)
+    radius_layers = polar_radius_layers(opts)
+
+    vl = append_layers(vl |> Vl.data_from_values(%{"_r" => [0]}), grid_layers ++ radius_layers)
 
-    if opts[:return_grid] do
-      unless opts[:radius_marks] do
-        raise ArgumentError, "missing :radius_marks option"
+    put_in(
+      vl.spec["_vl_polar_config"],
+      opts
+      |> Keyword.take([:radius_marks, :angle_offset, :direction])
+      |> Map.new()
+    )
+  end
+
+  def polar_plot(vl \\ Vl.new(), data, mark, fields, opts \\ []) do
+    opts = Keyword.validate!(opts, scheme: "turbo", hide_axes: false)
+
+    vl_polar_config =
+      case vl.spec do
+        %{"_vl_polar_config" => conf} -> Enum.to_list(conf)
+        _ -> [radius_marks: [1, 2, 3, 4, 5], angle_offset: 0, direction: :counter_clockwise]
```

Comment:
I think we should raise if there is no config.

Reply:
If there's no config, we can't do the standalone plot (the one where we get the cartesian-converted plot for polar data). Maybe we can require that either the config is present _or_ those three options are present.
If both are present, we could make it so that the config takes precedence.

Reply:
> If there's no config, we can't do the standalone plot (the one where we get the cartesian-converted plot for polar data).

That's fine by me. Let's raise for now. It is always easier to add features later.

Reply:
I managed to use the `fields` along with a check for the presence of the `:radius_marks` config to allow for the plot to still be used in a standalone fashion.

I'll commit it like this and we can remove the code if you feel it's bloat